osf/management/commands/check_ia_metadata.py

```diff
@@ -0,0 +1,104 @@
+import re
+import requests
+from django.core.management.base import BaseCommand
+from osf.models import Registration
+from website import settings
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class IAMetadataError(Exception):
+    def __init__(self, message=None, fields=None):
+        self.message = message
+        self.fields = fields
+
+
+get_ia_field = lambda field: Registration.IA_MAPPED_NAMES.get(field, field)
+
+mirrored_attrs = list(Registration.SYNCED_WITH_IA)
+from django.db.models import F
+
+mirrored_fields = mirrored_attrs + ['subjects', 'tags', 'affiliated_institutions']
+mirrored_relationships = [
+    'subjects__text',
+    'affiliated_institutions__name',
+    'tags__name',
+]
+
+
+def check_ia_metadata(collection=settings.IA_ROOT_COLLECTION, guids=None):
+    item_data = requests.get(
+        f'https://archive.org/advancedsearch.php'
+        f'?q=collection%3A({collection}) AND identifier:(osf-registrations-*-{settings.ID_VERSION})'
+        f'&fl[]='
+        + f'&fl[]='.join([get_ia_field(field) for field in mirrored_fields])
+        + f'&fl[]=identifier'
+        f'&rows=100000'
+        f'&output=json'
+    ).json()['response']['docs']
+    item_data = sorted(item_data, key=lambda x: x['identifier'])
+
+    archived_registrations = Registration.objects.filter(ia_url__isnull=False).exclude(ia_url='').order_by(
```

Comment:
When I ask Matt Clark to reset all the urls on staging so I can retry the archive process and he reasonably sets it to this. Obviously the edgiest of cases, but good in case of similar snafu in the future.